rln-research/2022-07-22-rln-relay-journey.md

```diff
@@ -0,0 +1,150 @@
+
+**Why**: The WAKU-RLN-RELAY project has been started more than two years ago to protect the open p2p messaging protocol of [WAKU-RELAY](https://rfc.vac.dev/spec/11/) against spam and DDoS attacks.
+However, the use case of WAKU-RLN-RELAY is not limited to the WAKU-RELAY protocol and can be adopted by any open p2p routing system.
+It is worth mentioning that WAKU-RELAY is a minor extension of [Libp2p Gossipsub protocol](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/README.md), hence the spam protection property of WAKU-RLN-RELAY is directly applicable to the libp2p Gossipsub protocol too. 
+In the nutshell, spam protection is achieved by limiting the messaging rate of every network participant so that no user can flood the network with a large number of messages and congest/DDoS relayers.
+One special focus of WAKU-RLN-RELAY is user privacy which is aligned with the WAKU-RELAY privacy objectives.
+That is, it does not rely on any personally identifiable information of users like IP address or peer ID to control their messaging rate.  
+This makes it superior in terms of privacy guarantees compared to the centralized system where users are asked to share some PII like phone number, or email address for network participation.
+Moreover, WAKU-RLN-RELAY is superior to its privacy-preserving counterparts i.e., proof-of-work in terms of performance.
+The computational overhead imposed by proof-of-work is not suitable for resource-limited devices like mobile phones whereas the computational overhead incurred by the WAKU-RLN-RELAY protocol is minimal and as of the time of this post is approximately less than half a second. 
+
+
+**How**: 
+Network participants which want to send messages in the network should be registered in the RLN group.
+The state of the group is moderated by a smart contract which holds a list of enrolled members.
+Each participant has a secret identity key and registers a commitment of that to the membership smart contract and deposits some funds.
+The key to withdraw the locked fund is the user's secret identity key, so if someone knows someone's secret identity key then it can take the associated fund.
+Messaging rate is enforced per some time interval called an epoch.
+For each epoch, the user sends a message by proving that it knows the opening to one of the commitments registered in the RLN group as well as it reveals a Shamir share of its secret id key. 
+The share is derived in such a way that two of such shares for the same epoch allow the construction of the corresponding secret id key. 
+The secret sharing has another element which is an application-defined identifier i.e., RLN identifier by which you can make the secret sharing unique to your application. 
+That is, if a user wants to use the same RLN membership credential for messaging in two or more different applications, then it can do so without being worried to get slashed by combining the shares of its identity key published in those applications.
```

Comment:
```suggestion
That is, if a user wants to use the same RLN membership credential for messaging in two or more different applications, then they can do so without being slashed by combining the shares of its identity key published in those applications.
```

Should there be a statement here on *why* users shouldn't use the same membership credentials for different applications? (anonymity guarantees)

Reply:
If users do not use the _same credentials_ in _different groups_, they should have no security concerns. What is pictured here is a different scenario and that is a user uses its _membership to one single rln group_ to message in _different applications_. Does that answer your question?

Reply:
Ah, I see! I was of the opinion that by application it implied a different membership group.